app/models/start_end_time_validator.rb

```diff
@@ -0,0 +1,6 @@
+class StartEndTimeValidator < ActiveModel::Validator
```

Comment:
The validator class should not reside in `app/models`!

Let's create `app/validators` folder and put the class into it.